src/stats.c

```diff
@@ -788,187 +789,349 @@ void  stats_updateMaxStats(TMaxStats maxStats[], int i, int j, double x)
 
 //=============================================================================
 
-double stats_getNodeStat(int index, int element)
+int stats_getNodeStat(int index, int paramtype, double* value)
 //
 // Input:    index
 //           element = element to return
 // Return:   value
 // Purpose:  Gets a Node Stat for toolkitAPI
 //
 {
-	double value;
+	int errorcode = 0;
 
-	switch (element)
+	// Check if Open
+	if (swmm_IsOpenFlag() == FALSE)
```

Comment:
The flow of control here is not what it should be. As soon as an error condition is encountered you want execution to stop and have the error returned. But this logic continues to check for errors. What it multiple error conditions are present? You are handing back information on the last error condition encountered not the first encountered. 

Handle the most severe error condition first with an `if` statement, then the next most severe with an `else if`, and so on until all error conditions have been checked, then use an `else` to enclose the block  of main functionality.